eZ/Publish/API/Repository/Tests/LocationServiceTest.php

```diff
@@ -1472,6 +1473,59 @@ public function testSwapLocation()
         );
     }
 
+    /**
+     * Test for the swapLocation() method with custom aliases.
+     *
+     * @see \eZ\Publish\API\Repository\LocationService::swapLocation()
+     * @depends eZ\Publish\API\Repository\Tests\LocationServiceTest::testLoadLocation
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\ForbiddenException
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
+     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException
+     */
+    public function testSwapLocationCustomAliases()
+    {
+        $repository = $this->getRepository();
+        $locationService = $repository->getLocationService();
+        $urlAliasService = $repository->getURLAliasService();
+        $this->createLanguage('pol-Pl', 'Polski');
+
+        $folder1 = $this->createFolder(['eng-GB' => 'Folder1', 'pol-Pl' => 'Folder1'], 2);
+        $folder2 = $this->createFolder(['eng-GB' => 'Folder2'], 2);
+        $location1 = $locationService->loadLocation($folder1->contentInfo->mainLocationId);
+        $location2 = $locationService->loadLocation($folder2->contentInfo->mainLocationId);
+
+        $urlAliasService->createUrlAlias($location1, '/custom-location1', 'eng-GB', false, true);
+        $urlAliasService->createUrlAlias($location1, '/custom-location1', 'pol-Pl', false, true);
+        $urlAliasService->createUrlAlias($location2, '/custom-location2', 'eng-GB', false, true);
+        $location1UrlAliases = $urlAliasService->listLocationAliases($location1);
+        $location2UrlAliases = $urlAliasService->listLocationAliases($location2);
+
+        $locationService->swapLocation($location1, $location2);
+        $location1 = $locationService->loadLocation($location1->contentInfo->mainLocationId);
+        $location2 = $locationService->loadLocation($location2->contentInfo->mainLocationId);
+        $location1UrlAliasesAfterSwap = $urlAliasService->listLocationAliases($location1);
+        $location2UrlAliasesAfterSwap = $urlAliasService->listLocationAliases($location2);
+
+        $urlAlias = new URLAlias([
+            'id' => $location1UrlAliases[0]->id,
+            'type' => $location1UrlAliases[0]->type,
+            'destination' => $location1UrlAliases[0]->destination,
+            'path' => $location1UrlAliases[0]->path,
+            'languageCodes' => ['eng-GB'],
+            'alwaysAvailable' => $location1UrlAliases[0]->alwaysAvailable,
+            'isHistory' => $location1UrlAliases[0]->isHistory,
+            'isCustom' => $location1UrlAliases[0]->isCustom,
+            'forward' => $location1UrlAliases[0]->forward,
+        ]);
+
+        self::assertEquals($folder1->id, $location2->contentInfo->id);
+        self::assertEquals($folder2->id, $location1->contentInfo->id);
+        self::assertNotEquals($location1UrlAliases, $location1UrlAliasesAfterSwap);
+        self::assertEquals($location2UrlAliases, $location2UrlAliasesAfterSwap);
```

Comment:
Location2 aliases should be same because language mask covered, but Location1 after a swap should be different because the new location does not have pol-PL language